common/src/main/java/org/keycloak/common/util/DelegatingSerializationFilter.java

```diff
@@ -0,0 +1,215 @@
+            this.createFilterMethod = createFilterMethod;
+        }
+
+        public Object getObjectInputFilter(ObjectInputStream ois) {
+            try {
+                return getObjectInputFilterMethod.invoke(null, ois);
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                LOG.warn("Could not read ObjectFilter from ObjectInputStream: " + e.getMessage());
+                return null;
+            }
+        }
+
+        public void setObjectInputFilter(ObjectInputStream ois, String filterPattern) {
+            try {
+                Object objectFilter = createFilterMethod.invoke(null, filterPattern);
+                setObjectInputFilterMethod.invoke(null, ois, objectFilter);
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                LOG.warn("Could not set ObjectFilter: " + e.getMessage());
+            }
+        }
+    }
+
+
+    static class EmptyFilterAdapter implements SerializationFilterAdapter {
+
+        @Override
+        public Object getObjectInputFilter(ObjectInputStream ois) {
+            return null;
+        }
+
+        @Override
+        public void setObjectInputFilter(ObjectInputStream ois, String filterPattern) {
+
+        }
+
+    }
+
+
+    // If codebase moves to Java 9+ could use Java 9+ classes directly without reflection and keep the old variant with reflection
+    static class OnJavaAfter8 implements SerializationFilterAdapter {
+
+        private static final Method getObjectInputFilterMethod;
+        private static final Method setObjectInputFilterMethod;
+        private static final Method createFilterMethod;
+
+        static {
+
+            Method getObjectInputFilter;
+            Method setObjectInputFilter;
+            Method createFilter;
+
+            try {
+                ClassLoader cl = Thread.currentThread().getContextClassLoader();
+                Class<?> objectInputFilterClass = cl.loadClass("java.io.ObjectInputFilter");
+                Class<?> objectInputFilterConfigClass = cl.loadClass("java.io.ObjectInputFilter$Config");
+                Class<?> objectInputStreamClass = cl.loadClass("java.io.ObjectInputStream");
+                getObjectInputFilter = objectInputStreamClass.getDeclaredMethod("getObjectInputFilter");
+                setObjectInputFilter = objectInputStreamClass.getDeclaredMethod("setObjectInputFilter", objectInputFilterClass);
+                createFilter = objectInputFilterConfigClass.getDeclaredMethod("createFilter", String.class);
+                LOG.info("Using OnJavaAfter8 serialization filter adapter");
+            } catch (ClassNotFoundException | NoSuchMethodException e) {
+                LOG.warn("Could not configure SerializationFilterAdapter: " + e.getMessage());
+                getObjectInputFilter = null;
+                setObjectInputFilter = null;
+                createFilter = null;
+            }
```

Comment:
Yes, it will be slightly better. Even though in theory it shouldn't happen in newer JDK versions as methods should be there, but I am not 100% sure...